python/lsst/obs/base/formatters/fitsExposure.py

```diff
@@ -628,7 +628,13 @@ def _fixFilterLabels(self, file_filter_label, should_be_standardized=None):
                 warnings.warn(
                     f"Data ID {self.dataId} is missing (implied) value(s) for {missing}; "
                     "the correctness of this Exposure's FilterLabel cannot be guaranteed. "
-                    "Call Registry.expandDataId before Butler.get to avoid this."
+                    "Call Registry.expandDataId before Butler.get to avoid this.",
+                    # It is not entirely obvious what the correct stacklevel
+                    # should be for a formatter. Outside of obs_base will be
+                    # the butler datastore. Outside of daf_butler might be
+                    # the right answer and could be the caller using butler.get
+                    # or could be a runQuantum method.
```

Comment:
Maybe I leave it warning from the formatter. @TallJimbo do you have an opinion on where a warning from a formatter should appear to be coming from?

Reply:
We definitely don't want to fully lose any of the traceback, but if we have any control over what users see most prominently, it's probably whatever low-level I/O code the formatter is calling (e.g. if it's coming from cfitsio, I'd want to see cfitsio's error message).

Reply:
This question is specifically about what line and file this warning is reported as coming from.  For a formatter that was not directly called by the user maybe the formatter is the right source of the warning message.  The only other alternative would be the place that called butler.get.

Reply:
Ah, sorry, didn't look closely enough.  For this particular warning, whatever called `QuantumContext.get` would be ideal.

Reply:
I could extend `find_outside_stacklevel` a bit. My initial thought was to takes `*args` and so use it here with:

```python
stacklevel=find_outside_stacklevel("lsst.obs.base", "lsst.daf.butler")
```
and then it would be reported as coming from `QuantumContext.get` in the pipeline execution environment. At least it would come from the user code if they called `butler.get()`.  Making it come from the caller of `QuantumContext.get` seems a bit trickier. One option is to add `lsst.pipe.base.QuantumContext` to the list as well and so be able to filter out classes. I can give that a try but it would have to be on a new ticket. For now then I'll merge this and then come back later to fix it up (I'll add a TODO ticket).

Reply:
Deferring is completely fine.  This particular warning should be sufficiently rare that if somebody asks on Slack every time they see it and I have to answer them it still won't be much of a burden, so on its own I don't think this is sufficient motivation for big enhancements to `find_outside_stacklevel` (though I agree that might provide useful elsewhere).